### ISSUE TITLE
Uniformative error message when missing a "self" while importing a module

### ISSUE THREAD

**Issue by julienw** [rust-lang/rust#31573]
Exemple code:
```rust
mod foo {
  mod hello {
    pub enum world {}
  }
  use hello::world;
}

fn main() {
}
```

Error is:
```
<anon>:6:7: 6:12 error: unresolved import `hello::world`. Maybe a missing `extern crate hello`? [E0432]
<anon>:6   use hello::world;
               ^~~~~
<anon>:6:7: 6:12 help: see the detailed explanation for E0432
error: aborting due to previous error
playpen: application terminated with error code 101
```

The compiler should detect that "self" is missing.

Note this is an error I do all the time as a beginner.

**Comment by SimonSapin** [comment]
Even without doing an analysis of what could have likely been intended this error message could mention that paths in `use` are absolute by default, unless they start with `self::` or `super::`.

**Comment by frewsxcv** [comment]
Ran into this a few days ago, would love to the messaging here improved :+1:

**Comment by birkenfeld** [comment]
The linked PR adds the information to the long diagnostics, at least.

Note that if you just have an `use module;` then the error message is markedly different:
```
error: unresolved import `modname`. There is no `modname` in the crate root [E0432]
```
I wonder if some consolidation would be in order.

**Comment by jseyfried** [comment]
Fixed in #36289.
